psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c

```diff
@@ -0,0 +1,234 @@
+//#include <project.h>
+
+#include "dht11.h"
+#include <stdint.h>
+#include <dht.h>
+
+#define null 0
+
+void delay(void);
+void unpack_dht_data(void);
+
+uint8_t check;
+uint8_t decode_flag;
+uint8_t decoded_temperature_data;
+uint8_t decoded_humidity_data;
+
+int temperature_values[8];
+int humidity_values[8];
+int diff_count_values[44]={0};
+int countvalues_datatimer[44]={0};
+int executed;
+int count;
+int risingedge_counts=0;
+
+//_CY_ISR_PROTO(Delay_Timer_ISR_Handler);
+//_CY_ISR_PROTO(DHT_Pin_ISR_Handler);
+
+//What Return_Decoded_Temperature does:
+//    -Returns the data in decoded_temperature_data
+// 
+//Parameters: None
+//Returns: uint8_t decoded_temperature_data
+
+uint8_t dht_get_temperature(){
+    
+    if(decoded_temperature_data == null){
+        return -1;
+    }
+    else{
+        return decoded_temperature_data; 
+    }    
+}    
+
+//What Return_Decoded_Humidity does:
+//    -Returns the data in decoded_humidity_data
+// 
+//Parameters: None
+//Returns: uint8_t decoded_humidity_data
+
+uint8_t dht_get_humidity(){
+    
+    if(decoded_humidity_data == null){
+        return -1;
+    }else{
+        return decoded_humidity_data;
+    }    
+}   
+
+
+//What delay_funct does:
+//    - Enable Delay_Timer_ISR to detect Delay_Tiemr Terminal Count interrupt
+//    - Start the Delay_Timer component.
+//    - This function does not use the CPU delay API, CyDelay(), but uses Delay_Timer to generate delay. 
+//
+//Parameters: None
+//Return: None
+
+void delay_funct(void){
+    count = 1;
+    executed = 0;
+    // Enable Delay_timer_ISR
+    Delay_Timer_ISR_Enable();
+    //Write 0 to Control_Reg_Delay_Timer to starting Delay_Timer component 
+    Control_Reg_Delay_Timer_Write(0);
+}    
+
+
+//What decode dht data does:
+//    - The value stored in countvalues_datatimer[] array, is used to decode the data of DHT output
+//    - The time difference between countvalues_datatimer[i] and countvalues_datatimer[i-1] is calculated
+//    - If the countvalues_datatimer[i] is approximately 70 counts, it is data 0, and if countvalues_datatimer[i] is approximately 120, it is data 1. 
+//
+//Parameters: None
+//Returns: None
+
+void decode_DHT_data(void){
+    int i;
+    for(i=0; i<44; i++){
```

Comment:
Prefer pre-increment to post-increment because it requires fewer instructions. (Although GCC may be clever enough to optimize that out nowadays...) I realize this is pretty nitpicky, but it's a good habit to get into.